### DISCUSSION TITLE
Working of the form as a livewire component

### DISCUSSION THREAD

**Discussion by Pavelask** [filamentphp/filament#13599] [Help]
Hello everyone, I encountered the following problem: when creating a form in resource files, the form display has a normal appearance, but when you create the form as a liveiwre component, there are problems with the work and display of the form.

<img width="862" alt="Снимок экрана 2024-07-17 в 00 24 56" src="https://github.com/user-attachments/assets/233d4f56-695f-4e39-9f77-e2d31aca4d2a">
this form is in the resource file

<img width="1269" alt="Снимок экрана 2024-07-17 в 00 16 40" src="https://github.com/user-attachments/assets/8604850f-01c0-4639-8788-af772988c8da">
And this form is in the livewire component file

<img width="1490" alt="Снимок экрана 2024-07-17 в 00 17 30" src="https://github.com/user-attachments/assets/9db21b83-dab5-4bd6-9208-feaa10f571a3">
And errors in the console

Laravel v10.48.16 (PHP v8.1.29)
filament/actions     v3.2.93    Easily add beautiful action modals to any Livewire component.
filament/filament   v3.2.93    A collection of full-stack components for accelerated Laravel app development.
filament/forms    v3.2.93    Easily add beautiful forms to any Livewire component.
filament/infolists    v3.2.93    Easily add beautiful read-only infolists to any Livewire component.
filament/notifications   v3.2.93    Easily add beautiful notifications to any Livewire app.
filament/support     v3.2.93    Core helper methods and foundation code for all Filament packages.
filament/tables       v3.2.93    Easily add beautiful tables to any Livewire component.
filament/widgets   v3.2.93    Easily add beautiful dashboard widgets to any Livewire component.
livewire/livewire    v3.5.4     A front-end framework for Laravel.

**Comment by leandrocfe** [comment]
Did you add $this->form->fill() in the mount method? Could you share the code?

**Reply by Pavelask** [reply]
`<?php

namespace App\Livewire;

use Filament\Forms\Components\MarkdownEditor;
use Filament\Forms\Components\RichEditor;
use Filament\Forms\Components\Textarea;
use Filament\Forms\Components\TextInput;
use Filament\Forms\Components\Toggle;
use Filament\Forms\Concerns\InteractsWithForms;
use Filament\Forms\Contracts\HasForms;
use Filament\Forms\Form;
use Livewire\Component;

class CreateNews extends Component implements HasForms
{
    use InteractsWithForms;

    public ?array $data = [];

    public function mount(): void
    {
        $this->form->fill();
    }

    public function form(Form $form): Form
    {
        return $form
            ->schema([
                TextInput::make('name')
                    ->label('TextInput')
                    ->maxLength(255)
                    ->columnSpanFull(),
                RichEditor::make('description')
                    ->label('RichEditor')
                    ->columnSpanFull(),
                MarkdownEditor::make('content')
                    ->label('MarkdownEditor')
                    ->columnSpanFull(),
                Textarea::make('description')
                    ->label('Textarea')
                    ->rows(7)
                    ->columnSpanFull(),
                TextInput::make('slug')
                    ->label('TextInput')
                    ->maxLength(255),
                TextInput::make('sort')
                    ->label('TextInput')
                    ->maxLength(255)
                    ->default(500),
                Toggle::make('is_visible')
                    ->label('OFF|ON')
                    ->onColor('red')
                    ->offColor('danger')
                    ->inline(false)
                    ->onIcon('heroicon-m-bolt')
                    ->offIcon('heroicon-m-user'),
            ])->columns(3)
            ->statePath('data');
    }

    public function create(): void
    {
        dd($this->form->getState());
    }

    public function render()
    {
        return view('livewire.create-news');
    }
}
`

**Reply by leandrocfe** [reply]
You have two `description` fields in the form

**Reply by Pavelask** [reply]
After fixing the problem remains the same

![Снимок экрана 2024-07-17 в 11 45 07](https://github.com/user-attachments/assets/8c3f9ad9-5653-4e95-961b-e2a70af6cc1e)
![Снимок экрана 2024-07-17 в 11 45 32](https://github.com/user-attachments/assets/aef4e032-66b2-4335-bb40-4ce7da3f9546)

**Reply by leandrocfe** [reply]
Are you able to share this project on github or create another to reproduce this issue?

**Reply by Pavelask** [reply]
https://github.com/Pavelask/Blog 

namesite/news - Form with errors and incorrect display of elements

**Comment by franckopheno** [comment]
j'ai eu le meme probleme ... apparemment t'as pas suivie suivie l'installation du package filament/form je t'invite a suivre rigoureusement les instructions presentes sur cette page [https://filamentphp.com/docs/3.x/forms/installation](url) ceci meme si t'as deja installer le package pour le panel ...une fois fait je crois que ton probleme sera resolu
